### PR TITLE
Add Julia 1.6 doc changes to CI

### PR DESCRIPTION
**Pull request by DhairyaLGandhi**
Test against 1.6

### CODE REVIEW

File: docs/src/data/onehot.md [hunk 1]
```diff
@@ -44,13 +44,13 @@ Flux.onecold
 julia> using Flux: onehotbatch
 
 julia> onehotbatch([:b, :a, :b], [:a, :b, :c])
-3×3 Flux.OneHotArray{3,2,Array{UInt32,1}}:
+3×3 Flux.OneHotArray{UInt32,3,1,2,Vector{UInt32}}:
```
Comment (darsnack) [inline docs/src/data/onehot.md:47]:
```suggestion
3×3 Flux.OneHotArray{3,2,Vector{UInt32}}:
```

**Comment by DhairyaLGandhi** [comment]
The docs failing don't make much sense, possible that is an artefact of Documenter needing to update to 1.6.

Oh well, the state of this PR is better than master, so anyone want to approve this/ have suggestions for changes? @darsnack ?

**Review by darsnack — APPROVED** [review]
Documenter fails anytime Base (or any package) changes what `show` prints out. Unfortunately, Julia 1.6 changed the `Vector`/`Matrix` printing which really screwed up pretty much every print out. For future reference, you can use https://juliadocs.github.io/Documenter.jl/stable/man/doctests/#Fixing-Outdated-Doctests to automate replacing the REPL output when it happens.

But looks like docs are passing now, thanks!

**Comment by DhairyaLGandhi** [comment]
I did do that locally, but the refs were screwed up there. Not sure if that was related to the state of machine which is a shared one.

bors r+

**Comment by DhairyaLGandhi** [comment]
Bk suggests there is scalar indexing happening. Why is that in the ci? That undermines the value of the ci imo.

**Comment by darsnack** [comment]
It's disabled for the GPU integration tests. Being triggered somewhere in the losses tests? I would guess it might be related to #1534. I can add an explicit GPU test for that case.

**Comment by DhairyaLGandhi** [comment]
Not sure why bors is unhappy. Merging for now.

**Pull request merged**